prereise/gather/demanddata/transportation_electrification/smart_charging.py

```diff
@@ -70,7 +70,8 @@ def get_constraints(individual, kwhmi, power, trip_strategy, location_strategy,
     :return: (*pandas.DataFrame*) -- a DataFrame adding the calculated constraints
         to an individual vehicle's data
     """
-    constraints_df = individual.copy()
+    constraints_df = individual
```

Comment:
Is that not a good thing to copy?

Reply:
I think unnecessary copying might be slowing things down in this code, I think it might potentially more performant to use views:
https://www.skytowner.com/explore/difference_between_copy_and_view_in_pandas

Reply:
Just saw in the `smart_charging` function that `individual` is updated in each iteration and hence no copy is needed